### DISCUSSION TITLE
router.query.id is undefined on first render after reloading a page

### DISCUSSION THREAD

**Discussion by Bruno0221** [neuefische/questions#825] [Web] [answered]
if I open an event from the homepage it works fine but when I then reload the page, on the first render, router.query.id is undefined, giving me a bad request for data fetching. this crashes the app sometimjes, sometimes it does not. If I reload the page, sometimes it renders, sometimes it does not. this is the branch with logs for debugging:
https://github.com/MareikeFla/community-app/tree/US_connect-comments-to-user
This issue is allready occuring in the main branch, so the error did not come up with the changes in this branch.

**Comment by onemanwenttomow** [comment, marked as answer]
In Next.js, router.query parameters are populated asynchronously. This means that on the initial render, especially after a page reload, `router.query.id` might be undefined because the router hasn't parsed the dynamic route yet.

That means the first request using SWR is to (for example) `https://community-app-topaz.vercel.app/api/events/undefined`

Normally i would suggest something like this

```js
const id = router.query.id;
const { data, error } = useSWR(id ? `/api/data/${id}` : null);
```

i.e. no request will be made if the value of `id` is undefined.

**Reply by Bruno0221** [reply]
That did the trick, thank you so much!!!

**Reply by onemanwenttomow** [reply]
So this looks like a new problem. 

The problem is that comments could be undefined.

There are several options for handling this. This could be a potential simple solution [to be used here](https://github.com/MareikeFla/community-app/blob/US_connect-comments-to-user/components/CommentSection/CommentSection.js). 

``` js
if (!fetchedComments.comments) {
    return <div>Loading comments...</div>; // or any other loading state representation
}
```
